Request announcement by cmpctblock AFTER requesting cmpctblock/blocktxn

rebroad opened the pull request:
This will speed up block propagation since the annoucement request can
wait until after we have requested the block data.

rebroad commented on src/net_processing.cpp:
```diff
@@ -1847,28 +1847,29 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                     req.blockhash = pindex->GetBlockHash();
                     connman.PushMessage(pfrom, msgMaker.Make(NetMsgType::GETBLOCKTXN, req));
                 }
+                if (!fAlreadyInFlight && mapBlocksInFlight.size() == 1 && pindex->pprev->IsValid(BLOCK_VALID_CHAIN)) {
```
Might also want to include a check that ret == true here

gmaxwell commented:
Is the theory here that it will make it faster because it will move the getdata ahead of the 12 byte sendcmpt message?

rebroad commented:
@gmaxwell yes

rebroad commented:
rebased on top of part of #9252

rebroad commented:
@gmaxwell Also worth bearing in mind that some implementations may round-robin the messages being processed from peers, so changing the order of the messages might mean being one place further forward in the queue for receiving blocks from such a peer.

rebroad committed:
Request announcement by cmpctblock AFTER requesting cmpctblock/blocktxn

This will speed up block propagation since the annoucement request can wait until after we have requested the block data.

rebroad commented:
what am I talking about?! since #3180 the bitcoin core implementation round-robins the messages being processed from peers - this this changing of order is more significant than I'd realised...

TheBlueMatt commented:
This would make a noticeable difference in some cases, not sure why I missed this until now.

TheBlueMatt commented:
Should be obsoleted by #9400.

fanquake commented:
Closing now that #9400 has been merged. If there are still changes here that you'd like reviewed, please rebase and reopen.
